Code/Source/VehicleDynamics/VehicleInputsState.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Time/ITime.h>
+#include <type_traits>
+//#include <concepts> //C++20
+
+namespace VehicleDynamics
+{
+    /* C++ 20
+    template<typename T>
+    concept Zeroable = std::integral<T> or std::floating_point<T>; // arithmetic types
+
+    template<typename T>
+    requires Zeroable<T> class InputZeroedOnTimeout
+    */
+
+    //! Inputs with an expiration date - effectively is zero after a certain time since update
+    template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+    class InputZeroedOnTimeout
+    {
+    public:
+        InputZeroedOnTimeout(int64_t timeoutUs = 200000)
+            : m_timeoutUs(timeoutUs)
+        {
+        }
+
+        void UpdateValue(T updatedInput)
+        {
+            m_input = updatedInput;
+            m_lastUpdateUs = GetTimeSinceStartupUs();
+        }
+
+        T GetValue() const
+        {
+            if (auto timeSinceStartUs = GetTimeSinceStartupUs(); timeSinceStartUs - m_lastUpdateUs > m_timeoutUs)
+            {
+                return 0;
```

Comment:
If 0 can be a valid value of an input maybe returning `std::pair<bool, T>` would be a better solution here. result.first would indicate if the value should be considered up to date.

Reply:
This could be the case if we had a controller that behaves differently in such cases, but for us a zero is a zero at the moment.
Note that when implementing a "sleeping" or "handbrake" behavior, this part of the code might be affected.